test/e2e/federated-service.go

```diff
@@ -168,8 +170,25 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 				}
 
 				createBackendPodsOrFail(clusters, nsName, FederatedServicePodName)
+
 				service = createServiceOrFail(f.FederationClientset_1_5, nsName, FederatedServiceName)
-				waitForServiceShardsOrFail(nsName, service, clusters)
+				obj, err := conversion.NewCloner().DeepCopy(service)
+				// Cloning shouldn't fail. On the off-chance it does, we should
+				// shallow copy service to serviceShard before failing.
+				if err != nil {
+					serviceCopy := *service
+					serviceShard = &serviceCopy
+					framework.ExpectNoError(err, fmt.Sprintf("Error in deep copying service %q", service.Name))
+				}
+				var ok bool
+				serviceShard, ok = obj.(*v1.Service)
+				if !ok {
```

Comment:
Same here. If the object is not castable to service then something is REALLY broken.